# 分数与选择标准实验

比较五名学生的平均分和选中分数。这是自行构造的百分制示例，不是真实学生记录。只使用 Python 标准功能。

## 输入

修改 `scores` 和 `threshold` 后，重新运行计算与输出单元。

In [1]:
scores = [82, 75, 45, 90, 61]
threshold = 60

In [2]:
if not scores:
    raise ValueError("scores must not be empty")

mean_score = sum(scores) / len(scores)
selected = [score for score in scores if score >= threshold]

In [3]:
print("count:", len(scores))
print("mean:", mean_score)
print("threshold:", threshold)
print("selected:", selected)

count: 5
mean: 70.6
threshold: 60
selected: [82, 75, 90, 61]


## 解释

默认平均值为 70.6。达到标准 60 的分数是 82、75、90 和 61，45 被排除。标准只影响选择，不改变整体平均值。

## 修改条件

1. 标准为 80 时，选择结果是 `[82, 90]`。
2. 把最后一个分数改为 100，平均值变为 78.4。
3. 空列表会产生 `ValueError`。错误后保留的旧输出不是新结果。

更新解释后，重启内核并从上到下运行。核对结果与解释后保存。